group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinator.java

```diff
@@ -266,6 +267,28 @@ CompletableFuture<OffsetDeleteResponseData> deleteOffsets(
         BufferSupplier bufferSupplier
     );
 
+    /**
+     * Complete a transaction. This is called when the WriteTxnMarkers API is called
```

Comment:
Is it worth specifying here that we call this for transactional offset commits on consumer offsets topics?

Reply:
That makes sense. I also added a check in `GroupCoordinatorService.completeTransaction` to fail if another partition is received as it is not expected.